.github/workflows/push.yml

```diff
@@ -44,25 +48,59 @@ jobs:
       - name: install dependencies Linux
         if: startsWith(matrix.os, 'ubuntu-')
         run: |
-          use_llvm_repos=0
+          use_llvm_repos=1
 
-          case "${{ matrix.os }}-${{ matrix.llvm }}" in
-            ubuntu-18.04-10) use_llvm_repos=1;;
-            ubuntu-18.04-11) use_llvm_repos=1;;
-            ubuntu-18.04-12) use_llvm_repos=1;;
-            ubuntu-18.04-13) use_llvm_repos=1;;
-            ubuntu-18.04-14) use_llvm_repos=1;;
-            ubuntu-20.04-13) use_llvm_repos=1;;
-            ubuntu-20.04-14) use_llvm_repos=1;;
-            *) use_llvm_repos=0;;
+          case "${{ matrix.os }}" in
+            ubuntu-18.04)
+              case "${{ matrix.llvm }}" in
+                8) use_llvm_repos=0;;
+                9) use_llvm_repos=0;;
+                10) use_llvm_repos=0;;
+                *) use_llvm_repos=1;;
+              esac
+              ;;
+            ubuntu-20.04)
+              case "${{ matrix.llvm }}" in
+                8) use_llvm_repos=0;;
+                9) use_llvm_repos=0;;
+                10) use_llvm_repos=0;;
+                11) use_llvm_repos=0;;
+                12) use_llvm_repos=0;;
+                *) use_llvm_repos=1;;
+              esac
+              ;;
+            ubuntu-22.04)
+              case "${{ matrix.llvm }}" in
+                11) use_llvm_repos=0;;
+                12) use_llvm_repos=0;;
+                13) use_llvm_repos=0;;
+                14) use_llvm_repos=0;;
+                15) use_llvm_repos=0;;
+                *) use_llvm_repos=1;;
+              esac
+              ;;
+            *) use_llvm_repos=1;;
```

Comment:
This defaults to using the LLVM repos _unless_ it's one of the versions packaged by Canonical. This should be a limited set, compared to the previous way which would need exceptions for every version _not_ packaged by Canonical.